packages/allure-hermione/src/index.ts

```diff
@@ -0,0 +1,318 @@
+    });
+    browser.addCommand("subSuite", async (testId: string, value: string) => {
+      await addLabel(testId, LabelName.SUB_SUITE, value);
+    });
+    browser.addCommand("owner", async (testId: string, value: string) => {
+      await addLabel(testId, LabelName.OWNER, value);
+    });
+    browser.addCommand("severity", async (testId: string, value: string) => {
+      await addLabel(testId, LabelName.SEVERITY, value);
+    });
+    browser.addCommand("tag", async (testId: string, value: string) => {
+      await addLabel(testId, LabelName.TAG, value);
+    });
+    browser.addCommand("issue", async (testId: string, name: string, url: string) => {
+      await addLink(testId, url, name, LinkType.ISSUE);
+    });
+    browser.addCommand("tms", async (testId: string, name: string, url: string) => {
+      await addLink(testId, url, name, LinkType.TMS);
+    });
+    browser.addCommand("attach", async (testId: string, source: string, mimetype: string) => {
+      await addAttachment(testId, source, mimetype);
+    });
+  });
+  hermione.on(hermione.events.NEW_WORKER_PROCESS, (worker) => {
+    // eslint-disable-next-line
+    // @ts-ignore
+    worker._process.on("message", (message) => {
+      if (message?.type === ALLURE_METADATA_CONTENT_TYPE) {
+        const { testId, metadata } = message as HermioneAttachmentMessage;
+
+        handleAllureAttachment(testId, metadata);
+      }
+    });
+  });
+  hermione.on(hermione.events.TEST_BEGIN, (test) => {
+    const { ALLURE_HOST_NAME, ALLURE_THREAD_NAME } = process.env;
+    const thread = ALLURE_THREAD_NAME || test.sessionId;
+    const hostname = ALLURE_HOST_NAME || os.hostname();
+    const currentTest = new AllureTest(runtime, Date.now());
+    const [parentSuite, suite, ...subSuites] = getSuitePath(test);
+
+    currentTest.name = test.title;
+    currentTest.fullName = test.fullTitle();
+    // TODO: do we need history ID here?
+    // currentTest.historyId = md5(currentTest.fullName);
+    currentTest.stage = Stage.RUNNING;
+
+    currentTest.addLabel(LabelName.HOST, hostname);
+    currentTest.addLabel(LabelName.LANGUAGE, "javascript");
+    currentTest.addLabel(LabelName.FRAMEWORK, "hermione");
+    currentTest.addLabel(LabelName.THREAD, thread);
+
+    if (parentSuite) {
+      currentTest.addLabel(LabelName.PARENT_SUITE, parentSuite);
+    }
+
+    if (suite) {
+      currentTest.addLabel(LabelName.SUITE, suite);
+    }
+
+    if (subSuites.length > 0) {
+      currentTest.addLabel(LabelName.SUB_SUITE, subSuites.join(" > "));
+    }
+
+    runningTests.set(test.id(), currentTest);
+  });
+  hermione.on(hermione.events.TEST_PASS, (test) => {
+    const currentTest = runningTests.get(test.id());
+
+    if (!currentTest) {
+      throw new Error("There isn't running test for the current session!");
+    }
+
+    currentTest.status = Status.PASSED;
+  });
+  // TODO: do we need to handle this?
+  // hermione.on(hermione.events.RETRY, (test) => {
```

Comment:
hmmm did retryed tests hsve its own `TEST_BEGIN` events? if not we totally need it

Reply:
Yes, as I see, each test passes all the cycle